view/frontend/templates/js/replacejs.phtml

```diff
@@ -406,6 +406,9 @@ if (!$block->isOnPageFromWhiteList() && !$block->isMinicartEnabled()) { return;
             });
         };
 
+        var isPaymentOnlyCheckout = function() {
```

Comment:
I think if we refactor something it's better to change across the code.
After this change sometimes we use `isPaymentOnlyCheckout()` and sometimes `getCheckoutType()`.
I don't know how important it is but function getCheckoutType() return checkout for page like http://m2.vitaliy.dev.bolt.me/checkout/ and allow mini cart work on this page.

Can you explain a little why you need this change?

Reply:
It's a fair point. I'll explain my logic and why I went with this route. 

getCheckoutType - onload of /checkout page the url will not always have a # value when this function runs for native checkout. This results in an inconsistant result where most of the time it is actually 'checkout' and not 'payment' so conditions which depends on 'payment' being there onload will not work. The only situation when 'payment' was returned as value was when I refreshed with # values in the url. 

The only reason I did not modify the existing function is I am concerned about additional items breaking if I were to change it. However I am not against modifying getCheckoutType to a similar condition where we check for this:
```javascript
        var getCheckoutType = function() {
            var trimmedUrl = trim(location.pathname, '/');
            return trimmedUrl === 'checkout' && trimmedUrl !== 'checkout/cart' && settings.quote_is_virtual  ? 'payment' : 'checkout';
        };
```

Reply:
We jumped on call and Mike explained me the fix. I think we can go ahead and update getCheckoutType function.
Can it be just
```
    var getCheckoutType = function() {
            return trim(location.pathname, '/') === 'checkout'   ? 'payment' : 'checkout';
        };
```

Reply:
Retesting with change.